compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/MockDesugar.java

```diff
@@ -102,15 +96,33 @@ public void generateMockFunctions(BLangPackage pkgNode) {
         Set<String> mockFunctionSet = mockFunctionMap.keySet();
 
         for (String function: mockFunctionSet) {
-            if (!function.contains("#")) { // Check added to skip desugaring for previous implementation mock functions
-                pkgNode.getTestablePkg().functions.add(generateMockFunction(function));
-            }
+            pkgNode.getTestablePkg().functions.add(generateMockFunction(function));
         }
     }
 
     private BLangFunction generateMockFunction(String functionName) {
-        // Get the orginal function from the pkgNode function list for reference
-        this.originalFunction = getOriginalFunction(functionName);
+        // Set the current mock object
+        this.mockFnObjectName = this.bLangPackage.getTestablePkg().getMockFunctionNamesMap().get(functionName);
+
+        // Reset both import and original functions
+        this.importFunction = null;
+        this.originalFunction = null;
+
+        // Function name is going to be 'ballerina/math:1.0.0 # absInt'
+        String currentPackageId = this.bLangPackage.packageID.toString();
+
+
+        // We need to identify if the function is part of the current package or imported package
+        if (functionName.contains(currentPackageId)) {
+            // Simply extract the name only and set it to function name
+            functionName = functionName.substring(functionName.indexOf(MOCK_ANNOTATION_DELIMITER) + 1);
+            this.originalFunction = getOriginalFunction(functionName);
+        } else {
+            // Extract the name and the package details
+            String packageName = functionName.substring(functionName.indexOf("/") + 1, functionName.indexOf(":"));
```

Comment:
```suggestion
            String packageName = functionName.substring(functionName.indexOf('/') + 1, functionName.indexOf(':'));
```
Faster `indexOf` method by using `char`

Reply:
Switched to using `char` instead of `String` for indexOf